src/schema/v2/object_identification.ts

```diff
@@ -156,9 +159,25 @@ const NodeField: GraphQLFieldConfig<any, ResolverContext> = {
           rootValueForChild(rootValue)
         )
       ).then((data) => {
-        // Add the already known type so `NodeInterface` can pluck that out in
+        let resolvedType = type
+
+        // If this is a union type, we need to return the specific member type
```

Comment:
@anandaroop I was curious about your comment on the union type implementing `resolveType` vs member types implementing `isTypeOf`.

I temporarily commented out every home view section except the artwork rail, and updated the union implementation from `resolveType` to `isTypeOf`.

Indeed, this did _not_ work - `resolveType` was `undefined` - but it seemed pretty easy to support that style as well!

Here's that attempt. This should work for either union-style membership definition - the union type implementing `resolveType` or member types implementing `isTypeOf`.